web/web.go

```diff
@@ -526,38 +529,193 @@ func dropForeignKey(w http.ResponseWriter, r *http.Request) {
+	json.NewEncoder(w).Encode(sessionState.conv)
+}
+func addIndexes(w http.ResponseWriter, r *http.Request) {
+	table := r.FormValue("table")
+	reqBody, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Body Read Error : %v", err), http.StatusInternalServerError)
+	}
+
+	newIndexes := []ddl.CreateIndex{}
+	if err = json.Unmarshal(reqBody, &newIndexes); err != nil {
+		http.Error(w, fmt.Sprintf("Request Body parse error : %v", err), http.StatusBadRequest)
+		return
+	}
+	//check new name for spanner name validity
+	newNames := getNameSlice(newIndexes)
+	if validity, invalidNewNames := checkSpannerNamesValidity(newNames); !validity {
+		http.Error(w, fmt.Sprintf("New name validation fails for following : %v", invalidNewNames), http.StatusBadRequest)
 		return
 	}
-	if !isUniqueName(newName) {
-		http.Error(w, fmt.Sprintf("New name %s is used before in the schema", newName), http.StatusBadRequest)
+
+	//check new name for non usage before in another table, foreign key constraint or index
+	if status, err := canRename(newNames, table); !status {
+		http.Error(w, err.Error(), http.StatusBadRequest)
 		return
 	}
-	sp.Fks[position].Name = newName
+
+	sp := sessionState.conv.SpSchema[table]
+	sp.Indexes = append(sp.Indexes, newIndexes...)
+
 	sessionState.conv.SpSchema[table] = sp
 	updateSessionFile()
 	w.WriteHeader(http.StatusOK)
 	json.NewEncoder(w).Encode(sessionState.conv)
 }
 
+func getValuesAsLowerCaseSlice(input map[string]string) []string {
+	output := []string{}
+	for _, value := range input {
+		output = append(output, strings.ToLower(value))
+	}
+	return output
+}
+
+func getNameSlice(input []ddl.CreateIndex) []string {
+	output := make([]string, len(input))
+	for _, value := range input {
+		output = append(output, value.Name)
+	}
+	return output
+}
+
+func checkSpannerNamesValidity(input []string) (bool, []string) {
+	status := true
+	invaliNewNames := []string{}
+	for _, newName := range input {
+		if _, status := internal.FixName(newName); status {
+			status = false
+			invaliNewNames = append(invaliNewNames, newName)
+		}
+	}
+	return status, invaliNewNames
+}
+
+func canRename(names []string, table string) (bool, error) {
+
+	namesMap := map[string]bool{}
+	//Check for this name isn't used by another table
+	for _, name := range names {
+		namesMap[name] = true
+		if _, ok := sessionState.conv.SpSchema[name]; ok {
+			return false, fmt.Errorf("new name : '%s' is used by another table", name)
+		}
+	}
+
+	//Check for this name isn't used by another foreign key
+	for _, sc := range sessionState.conv.SpSchema {
+		for _, foreignKey := range sc.Fks {
+			if sc.Name != table {
+				if _, ok := namesMap[foreignKey.Name]; ok {
+					return false, fmt.Errorf("new name : '%s' is used by another foreign key in table : '%s'", foreignKey.Name, sc.Name)
+				}
+			}
+		}
+	}
+
+	//This name isn't used by another index: you'll need to iterate over conv.SpSchema and check the Indexes of each CreateTable.
```

Comment:
Comments are typically written like instructions. For example, you can drop "you'll need to" here.